### DISCUSSION TITLE
How do I get data from $security.parseJWT(token, verificationKey) function from JSVM ?

### DISCUSSION THREAD

**Discussion by jahidhsn002** [pocketbase/pocketbase#3782] [Q&A] [answered]
2 function from JSVM on myhook.pb.js in hooks folder

```
  $security.createJWT(payload, signingKey, secondsDuration)
  $security.parseJWT(token, verificationKey)
```

```
  signingKey = "someexamplekey"
  const jwtToken = $security.createJWT({
    "avatar": "google_Ds3eQfYntb.png",
    "designation": "Managing Director",
    "disabled": false,
    "email": "ashiq@gmail.com",
    "employeeId": "ION-01",
    "isAdmin": false,
    "name": "Ashiq Haque",
    "verified": false
  },  signingKey, 360000)

  // Token Works on https://jwt.io/#debugger-io with key
  console.log(jwtToken)

  const data = $security.parseJWT(jwtToken, signingKey)
  console.log(JSON.stringify(data))
```

but data returns as empty object. {} .... may be its related to verificationKey. How do I parse data from JWT token ?

**Comment by ganigeorgiev** [comment]
Hm, `JSON.stringify` doesn't work in this case because the returned Go type doesn't have public fields and/or is a [map with methods](https://pkg.go.dev/github.com/dop251/goja#hdr-Maps_with_methods).

I'll consider adding a method for this (or wrap the return result of the parse methods) for the next release, but at the moment I can't think of an easy workaround without modifying the Go bindings.

Could you elaborate a little more on your use case and how you plan to use the above methods?

If you are trying to generate an auth record token, I recommend using the builtin [`$tokens.recordAuthToken($app, record)`](https://pocketbase.io/jsvm/functions/_tokens.recordAuthToken.html) helper.

For the reverse, aka. locating an auth record by a token (and validating it) you can use:
```js
const record = $app.dao().findAuthRecordByToken(token, $app.settings().recordAuthToken.secret)
```

**Comment by jahidhsn002** [comment, marked as answer]
For now resolved it with ['jwt-decode'](https://github.com/auth0/jwt-decode).  For me this is resolved.

```
const jwtDecode = require(`${__hooks}/jwt-decode.js`)
try {
    // For verifying token only
    $security.parseJWT(jwtToken, signingKey)

    // Decoded data
    console.log(jwtDecode(jwtToken))
} catch {}
```

but I think $security.parseJWT(token, verificationKey) this should not return {}. Thanks for the help by the way.
